git/git.go

```diff
@@ -87,13 +87,20 @@ type Commit struct {
 	Encoded []byte
 	// Message is the commit message, containing arbitrary text.
 	Message string
+	// ShowFullRef indicates whether the string representation of the commit
+	// should include the entire reference.
+	ShowFullRef bool
 }
 
-// String returns a string representation of the Commit, composed
-// out the last part of the Reference element, and/or Hash.
+// String returns a string representation of the Commit.
+// By default its composed out the last part of the Reference element
+// and/or Hash. If c.ShowFullRef=true, the entire Reference is used.
 // For example: 'tag-1/a0c14dc8580a23f79bc654faa79c4f62b46c2c22',
 // for a "tag-1" tag.
 func (c *Commit) String() string {
+	if c.ShowFullRef {
```

Comment:
Instead of making this an option on the commit, I think it should be a separate method on the commit itself. Otherwise the `Commit` all of sudden bears other data than just commit data.

Reply:
I considered that as well, but went ahead with this approach as I didn't want consumers of the package having to decide whether to call `.String()` or `.FullString()` (for example)

Reply:
I think that's absolutely up to the consumer to decide, and dictating it from the perspective of the library is wrong.